# <center>***Exploratory Analysis of Ireland Weather***<center>


*********************************


![WEATHER](https://i.pinimg.com/564x/1f/8e/8f/1f8e8f16b0be992424f3f5b849706e8b.jpg)

##### Author: Gabriela Domiciano Avellar
##### Project: Programming for Data Analytics




## Project: Analysis of Weather Forecast Data in Ireland

This project showcases how command-line tools and  scripting, and automation can be utilized to manage and process weather forecast data in Ireland. The data was coleceted from [Historical Data](https://www.met.ie/climate/available-data/historical-data) and  [Today's Weather Athenry](https://data.gov.ie/dataset/todays-weather-athenry), The goal is to understand and analyze forecast data, providing detailed climate insights.


******************************

In [13]:
# Exploratory Analysis: 'nui_galway.csv'
# Import required packages:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

In [14]:

# Let's take a look at the Dataset.
nui_df = pd.read_csv('/Users/gabrieladomicianoavellar/Desktop/PFDA_project/data/nui_galway.csv')
print(nui_df)

       Year  Month  Day  Max (°F)  Min (°F)  Max (°C)  Min (°C)
0      1851      1    1       NaN       NaN       NaN       NaN
1      1851      1    2       NaN       NaN       NaN       NaN
2      1851      1    3       NaN       NaN       NaN       NaN
3      1851      1    4       NaN       NaN       NaN       NaN
4      1851      1    5       NaN       NaN       NaN       NaN
...     ...    ...  ...       ...       ...       ...       ...
41937  1965     10   27       NaN       NaN      14.8      12.1
41938  1965     10   28       NaN       NaN      13.0       8.6
41939  1965     10   29       NaN       NaN      14.0       6.8
41940  1965     10   30       NaN       NaN      11.4       9.0
41941  1965     10   31       NaN       NaN      14.3       7.0

[41942 rows x 7 columns]


In [15]:
# Calculate descriptive statistics, and saves it to a CSV file called 'summary.csv'
nui_df.describe().T.to_csv('data/summary.csv')

- **Year**: Data covers the period 1851 to 1965.
- **Month**: Includes all months.
- **Day**: Covers days 1 to 31.
- **Max (°F)**: Temperatures range from 4°F to 89°F.
- **Min (°F)**: Temperatures range from 11°F to 77°F.
- **Max (°C)**: Highs from -15.6°C to 31.7°C.
- **Min (°C)**: Lows from -11.7°C to 25.0°C.


In [25]:

Low_df = pd.DataFrame(nui_df)

# Removing rows with missing values in the minimum temperature column
df_cleaned = Low_df.dropna(subset=["Min (°C)"])

# Finding the row with the lowest minimum temperature
temp_min_row = df_cleaned.loc[df_cleaned["Min (°C)"].idxmin()]

# Resultado
lowest_temp_year = temp_min_row["Year"]
lowest_temp_month = temp_min_row["Month"]
lowest_temp_value_c = temp_min_row["Min (°C)"]

print(f"The lowest temperature was {lowest_temp_value_c:.2f}°C, registered in the month {lowest_temp_month} of the year {lowest_temp_year}.")


The lowest temperature was -11.70°C, registered in the month 1.0 of the year 1945.0.


In [26]:
Low_df = pd.DataFrame(nui_df)

# Removing rows with missing values in the minimum temperature column
df_cleaned = Low_df.dropna(subset=["Max (°C)"])

# Finding the row with the highest minimum temperature
temp_max_row = df_cleaned.loc[df_cleaned["Max (°C)"].idxmax()]

# Resultado
highest_temp_year = temp_max_row["Year"]
highest_temp_month = temp_max_row["Month"]
highest_temp_value_c = temp_max_row["Min (°C)"]

print(f"The highest temperature was {highest_temp_value_c:.2f}°C, registered in the month {highest_temp_month} of the year {highest_temp_year}.")


The highest temperature was 15.00°C, registered in the month 7.0 of the year 1921.0.
